### ISSUE TITLE
Length argument for renpy.input not enforced on Android

### ISSUE THREAD

**Issue by Org-gith** [renpy/renpy#3667]
(moved it from the crash android topic to there, because it felt out of place)
Retested the renpy.input on 7.5.0.22061301n, and it does work, but what seems to be the actual finicky issue with android input is the length argument, for example:

`persistent.playern = renpy.input("", default='', allow="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", exclude=None, length=8, with_none=None, pixel_width=None)`

The length argument doesn't seem to be enforced correctly on Android, I used to input a random name to get to the testing fast enough, not really looking at it, and that name ended up trespassing the limit set (which shouldn't be possible) so it didn't let me press enter to validate the input.
Then tested it on PC and the character limit is correctly enforced on it, and I wouldn't be able to input more than what the length argument says. The allow argument doesn't seem to be enforced on Android too.

(In case that's something, wasn't able to actually build for android on the very latest nightly, it returned this error):
FAILURE: Build failed with an exception.
* Where:
Build file 'C:\Users\alcia\Desktop\Beyond The Veil\renpy-7.5.0-sdk\rapt\project\app\build.gradle' line: 24
* What went wrong:
A problem occurred evaluating project ':app'.
> path may not be null or empty string. path='null'

**Comment by renpytom** [comment]
I'm unable to repeat any of this. I'd suggest waiting for the next prerelease,  and reinstalling from scratch. Feel free to reopen this issue if it's still broken then.

**Comment by Org-gith** [comment]
I'm starting to believe I'm seeing ghosts, understood I'll test it back when the prerelease is up then.

**Comment by Org-gith** [comment]
I tested it again and I really wish I could say it was fixed on my phone, I'm puzzled by this but I can still input any characters I want on android regardless of the length argument. I'm starting to think this could be a phone thing, it's very strange, this is what I get on a Samsung S10e running android 11:
![0](https://user-images.githubusercontent.com/78823063/174499213-1a9200bc-27a7-47cf-9448-2a8dbd417a79.jpg)

For the 'in case that's something' bit, this happened because I didn't sign any keys for android, so totally my bad (replicated the problem before fixing it by pressing yes when prompted to create a key both times. the first time I said no to both and it didn't let me build which is normal I think.)

**Comment by Org-gith** [comment]
I have another older android phone lying around, so I'll test on it and see if it works on it.

**Comment by Org-gith** [comment]
It was actually fixed by the latest update from Samsung, I tried again the same old build, and it works now after updating.
I use this Samsung phone only for Android testing, so I don't use it much and I missed some updates from Samsung. The phone updated at one point to the last Samsung update when I rebooted it before trying the fresh template build I made so I thought the template was behaving differently than my project since it had no problems, and I was very confused by this, and just now after 2 hours, I tried back the initial build that gave me the problem and now it doesn't do it anymore at all. And I changed literally nothing outside of the Samsung update happening. Such a waste of time, I apologize, so all along it was my phone requiring the latest update to not make this issue, not a Ren'py problem at all.
